src/systems/material.js

```diff
@@ -146,7 +146,7 @@ module.exports.System = registerSystem('material', {
     setTextureProperties(texture, data);
 
     // if we're on iOS, and the video is HLS, we currently need to do some hacks
-    if (utils.device.isIOS() && isHLS(videoEl)) {
+    if (utils.device.isIOS() && isHLS(videoEl.src || videoEl.getAttribute('src'), videoEl.type || videoEl.getAttribute('type'))) {
```

Comment:
no, i thought you were saying the substitution had already been done, so only changing the flag was necessary.  if the function has to be monkey patched, not sure it is any clearer to do it as you suggest rather than directly manipulating to return desired value